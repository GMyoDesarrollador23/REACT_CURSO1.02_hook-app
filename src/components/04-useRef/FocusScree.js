import React, { useRef } from "react";
import "../02-useEffect/effects.css";

export const FocusScree = () => {
   const inputRef = useRef();

   const handleClick = () => {
      inputRef.current.select();
      console.log(inputRef);
   };

   return (
      <div>
         <h1>Focus Scree</h1>
         <hr />
         <input
            ref={inputRef}
            className="form-control"
            placeholder="su nombre"
         />
         <button
            className="btn btn-outline-success mt-5"
            onClick={handleClick}
         >
            focus
         </button>
         c
      </div>
   );
};
