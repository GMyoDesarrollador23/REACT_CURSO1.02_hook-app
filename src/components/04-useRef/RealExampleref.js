import React, { useState } from "react";

import { MultiplesCustomHook } from "../03-examples/MultiplesCustomHook";

import "../02-useEffect/effects.css";

export const RealExampleref = () => {
   const [show, setShow] = useState(false);
   return (
      <div>
         <h1>RealExampleref</h1>
         <hr />

         {show && <MultiplesCustomHook />}

         <button
            className="btn btn-primary"
            onClick={() => {
               setShow(!show);
            }}
         >
            {show ? <div>hide</div> : <div>show</div>}
         </button>
      </div>
   );
};
