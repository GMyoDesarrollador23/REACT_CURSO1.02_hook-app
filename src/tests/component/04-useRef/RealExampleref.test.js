import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";

import { RealExampleref } from "../../../components/04-useRef/RealExampleref";

describe("Pruebas en <RealExampleref/>", () => {
   const wrapper = shallow(<RealExampleref />);
   
   // ----------------------------------------------------
   test("debe de mostrarso correctamente", () => {
      expect(wrapper).toMatchSnapshot();

      expect(
         wrapper.find("MultiplesCustomHook").exists()
      ).toBe(false);
   });

   // ----------------------------------------------------
   test("debe de mostrar el componente <RealExampleref/>", () => {
      wrapper.find("button").simulate("click");

      expect(
         wrapper.find("MultiplesCustomHook").exists()
      ).toBe(true);
   });
   // ----------------------------------------------------
});
