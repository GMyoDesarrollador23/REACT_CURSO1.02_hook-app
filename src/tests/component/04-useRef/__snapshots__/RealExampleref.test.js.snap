// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RealExampleref/> debe de mostrarso correctamente 1`] = `
<div>
  <h1>
    RealExampleref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    <div>
      show
    </div>
  </button>
</div>
`;
